src/route/dataset-route.ts

```diff
@@ -0,0 +1,176 @@
+                }
+            ]
+        });
+        return;
+    }
+    const dataset = Dataset.createDataset(internalName, 'BetaUser');
+    const saved_dataset_record = await dataset.save();
+    const uploadDTO = await uploadCSV(req.file?.buffer, saved_dataset_record);
+    if (!uploadDTO.success) {
+        res.status(400);
+    }
+    res.json(uploadDTO);
+});
+
+apiRoute.get('/', async (req, res) => {
+    const datasets = await Dataset.find();
+    const fileList: FileDescription[] = [];
+    for (const dataset of datasets) {
+        fileList.push({
+            internal_name: dataset.internalName,
+            id: dataset.id
+        });
+    }
+    res.json({ filelist: fileList });
+});
+
+apiRoute.get('/:dataset', async (req, res) => {
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... file is null or undefined' });
+        return;
+    }
+    const dataset = await Dataset.findOneBy({ id: datasetID });
+    if (!dataset) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... dataset id is invalid' });
+        return;
+    }
+    const datafiles = await dataset.datafiles;
+    if (datafiles.length < 1) {
+        res.status(404);
+        res.json({ message: 'Dataset has no datafiles attached' });
+        return;
+    }
+    const dto = await datasetToDatasetDTO(dataset);
+    res.json(dto);
+});
+
+apiRoute.get('/:dataset/csv', async (req, res) => {
+    const dataLakeService = new DataLakeService();
+    const datasetID = req.params.dataset;
+    if (datasetID === undefined || datasetID === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... You must specify a dataset ID' });
+        return;
+    }
+    const dataset = await Dataset.findOneBy({ id: datasetID });
+    if (dataset === undefined || dataset === null) {
+        res.status(404);
+        res.json({ message: 'Dataset not found... Dataset ID not found in Database' });
+        return;
+    }
+    const datafiles = await dataset.datafiles;
+    const fileToDownload: Datafile | undefined = datafiles
+        .sort(
+            (first: Datafile, second: Datafile) =>
+                new Date(second.creationDate).getTime() - new Date(first.creationDate).getTime()
+        )
+        .shift();
+    if (fileToDownload === undefined || fileToDownload === null) {
+        res.status(404);
+        res.json({ message: 'Dataset has no file attached' });
+        return;
+    }
+    const file = await dataLakeService.downloadFile(`${fileToDownload.id}.csv`);
+    if (file === undefined || file === null) {
+        res.status(404);
+        res.json({ message: 'File not found... file is null or undefined' });
+        return;
+    }
+    res.setHeader('Content-Length', file.length);
+    res.setHeader('Content-Type', 'text/csv');
```

Comment:
Set this based on the file metadata and preserve the mime-type the user sent us when they uploaded it.

Reply:
As they can only upload CSVs currently I felt it was pretty save to explicitly say we're sending them back a CSV.  Additionally, this route is only designed to return back something in CSV form.  I envision a transformation step where things are bought together and a new CSV file is produced rather than just returning whatever happens to be in the datalake.